app/models/concerns/taxonomix.rb

```diff
@@ -78,33 +78,30 @@ def taxable_ids(loc = which_location, org = which_organization, inner_method = w
       # taxonomies. This value should only be returned for admin users.
       return nil if loc.nil? && org.nil? && User.current.try(:admin?)
 
+      inner_ids = []
       if SETTINGS[:locations_enabled] && loc.present?
-        inner_ids_loc = if Location.ignore?(self.to_s)
-                          self.pluck("#{table_name}.id")
-                        else
-                          inner_select(loc, inner_method)
-                        end
+        inner_ids |= if Location.ignore?(self.to_s)
+                       self.pluck("#{table_name}.id")
+                     else
+                       inner_select(loc, inner_method)
+                     end
       end
       if SETTINGS[:organizations_enabled] && org.present?
-        inner_ids_org = if Organization.ignore?(self.to_s)
-                          self.pluck("#{table_name}.id")
-                        else
-                          inner_select(org, inner_method)
-                        end
+        inner_ids |= if Organization.ignore?(self.to_s)
+                       self.pluck("#{table_name}.id")
+                     else
+                       inner_select(org, inner_method)
+                     end
       end
-      inner_ids = inner_ids_loc & inner_ids_org if (inner_ids_loc && inner_ids_org)
```

Comment:
The code in here is not equivalent to this. This line is getting the intersection of loc IDs and org IDs, so that it'll return IDs for stuff that belong in both things. The code in the PR is merely adding new objects through `|=` and so it never does this intersection. 

e.g: I call `Environment.taxable_ids(loc1, org1)` - using your method, I'll get Environments that have loc1 but no organization (wrong, only admin should see that). Using the original code you get Environments strictly in loc1 and org1.